### PR TITLE
Improve the about page & misc website enhancements

### PR DESCRIPTION
**Pull request by johnnovak**
# Description

This PR improves the following things on the website:

- Top-level sections get their own links. E.g. `$BASE_URL/getting-started/` is now a valid link; previously, this resulted in a 404, and we had to use `$BASE_URL/getting-started/introduction/` everywhere (the URL of the first sub-section). Same deal for `/releases/` which was `/releases/windows/` (Linux folks, rejoice 😎).

- I have updated the About page to more accurately document the project team's understanding of the scope, goals, and non-goals of DOSBox Staging based on our various discussions throughout the last year or so. To pre-empt any misunderstandings:

    **I am not arbitrarily making up some new goals here or changing the direction of the project!**

    This should reflect and document our common understanding (as I understand it 😅). If I left out or misrepresented something important, or you disagree with anything, please let me know, fellow project team members 😎 

- Added a Disclaimer and Data privacy policy section—we talked about these as well previously. Not so much to cover our asses, more to prevent so disaster if people are misguided enough to use Staging for "mission critical" work 😐 

You can view the new About page on the preview website:

https://dosbox-staging.github.io/preview/dev/about/

# Manual testing

Local testing & the preview website.

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [ ] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [ ] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: website/docs/about/index.md [hunk 1]
```diff
@@ -18,58 +18,55 @@ Because **DOSBox Staging** comes with sensible defaults, you'll need to write
...
 
 ## Non-goals
 
-- Support old operating systems (Windows 9x/Me/XP, OS/2, and Mac OS X 10.5)
-  and limited CPU/memory hardware, which are constraints
-  [DOSBox](https://www.dosbox.com/) continues to support.
+- Support old, end-of-life operating systems (e.g., Windows 7 or older, OS/2,
+  or Mac OS X 10.5 or older) and limited CPU/memory hardware, which are
+  constraints the original [DOSBox](https://www.dosbox.com/) continues to
+  support.
```
Comment (kklobe) [inline website/docs/about/index.md:55]: do we want to point to anything well-defined, like https://endoflife.date/windows, or leave it more at the team's discretion?

Comment (johnnovak) [reply website/docs/about/index.md:55]: I'd say the latter.

**Comment by kklobe** [comment]
The [Releases](https://dosbox-staging.github.io/preview/dev/releases/index.md) and [Getting Started](https://dosbox-staging.github.io/preview/dev/getting-started/index.md) links along the top of the About page give 404.

File: website/docs/about/index.md [hunk 2]
```diff
@@ -18,58 +18,55 @@ Because **DOSBox Staging** comes with sensible defaults, you'll need to write
...
 
 - Support the use of Windows 9x/Me in the emulator. Windows 9x/Me emulation
-  is supported by projects such as [QEMU](https://www.qemu.org) and
-  [VirtualBox](https://www.virtualbox.org/), although DOSBox forks like
-  [DOSBox-X](https://www.dosbox-x.com/) and [DOSBox
-  Pure](https://github.com/schellingb/dosbox-pure) also support this. This
-  may change if wanted by the DOSBox Staging community in general.
+  is supported by projects such as [QEMU](https://www.qemu.org),
+  [VirtualBox](https://www.virtualbox.org/) and
+  the [DOSBox-X](https://www.dosbox-x.com/) and [DOSBox Pure](https://github.com/schellingb/dosbox-pure) forks. This may change if the DOSBox Staging community wants it in general.
```
Comment (FeralChild64) [inline website/docs/about/index.md:59]: This line is, IMHO, too long.

**Review by japsmits — DISMISSED** [review]
Looks quite reasonable to me.

**Comment by johnnovak** [comment]
> The [Releases](https://dosbox-staging.github.io/preview/dev/releases/index.md) and [Getting Started](https://dosbox-staging.github.io/preview/dev/getting-started/index.md) links along the top of the About page give 404.
> 
> Ran a check with [LinkChecker](https://linkchecker.github.io/linkchecker/), `linkchecker --check-extern https://dosbox-staging.github.io/preview/dev/about/`

Ah, good catch! I forget to check in the two dummy `index.md` files to make the redirects work. Fixed & regenereated the preview site—no more broken links!

File: website/docs/about/index.md (hunk 2 above)
Comment (johnnovak) [reply website/docs/about/index.md:59]: Hmm, why did you single out this one? There are others that are almost twice as long 😆 

Anyway, how would you rephrase it?

**Comment by kklobe** [comment]
> > The [Releases](https://dosbox-staging.github.io/preview/dev/releases/index.md) and [Getting Started](https://dosbox-staging.github.io/preview/dev/getting-started/index.md) links along the top of the About page give 404.
> > Ran a check with [LinkChecker](https://linkchecker.github.io/linkchecker/), `linkchecker --check-extern https://dosbox-staging.github.io/preview/dev/about/`
> 
> Ah, good catch! I forget to check in the two dummy `index.md` files to make the redirects work. Fixed & regenereated the preview site—no more broken links!

The Mastodon link is still complaining (not sure if it's a valid complaint?):

```console
URL        `https://corteximplant.com/@DOSBox_Staging'
Name       `Mastodon'
Parent URL https://dosbox-staging.github.io/preview/dev/about/, line 1625, col 8
Real URL   https://corteximplant.com/@DOSBox_Staging
Check time 60.162 seconds
Result     Error: ReadTimeout: HTTPSConnectionPool(host='corteximplant.com', port=443): Read timed out. (read timeout=60)
```

**Comment by johnnovak** [comment]
> The Mastodon link is still complaining (not sure if it's a valid complaint?):

Different issue. Now the Mastodon link works for me, the service must have been down temporarily.

Thanks for being thorough 🏅 😄

**Review by shermp — DISMISSED** [review]
Content changes look ok to me. I especially like the disclaimers.

**Comment by johnnovak** [comment]
Okay, I've simplified the longest bullet point that mentioned PCem, 86box, etc. I think that's the one that @FeralChild64 really meant.

**Review by japsmits — APPROVED** [review]

**Comment by johnnovak** [comment]
FYI, these updates will go live with the v0.81.1 patch.

**Comment by johnnovak** [comment]
> > I've simplified the longest bullet point that mentioned PCem, 86box, etc.
> 
> I wondered what's the meaning of the bold text "Pursue hardware accuracy when it impacts the above goals or **detracts from the source material**".
> 
> And while now it's too late, I had a suggestion:
> 
>     * Leverage **upstream** and **community developments** in DOSBox.
> 
>     * Encourage **new contributors** by removing barriers to entry.
> 
>     * Fix, clean up, and integrate notable **community-developed patches**.
> 
> 
> Isn't the second "community development" redundant?

I was wondering about that myself, yeah, but they're a bit different. Anyway, this will do for now I think, it's good enough.

**Comment by johnnovak** [comment]
> **detracts from the source material**

I think that meant going a bit too far into emulating things, e.g., emulating the noise of the Sound Blasters 😄

**Comment by johnnovak** [comment]
> And I'm one of the supporters of HDD noise emulation (it was like a constant background PC status briefing + heads-up warning system for some game events)

Yeah, I actually liked that too and I find it nostalgic. Not sure how practical it is an emulator like DOSBox, though, because you'd need to emulate the drive heads moving on the platters, etc. Bit of an overkill 😅

**Comment by johnnovak** [comment]
> > how practical it is an emulator like DOSBox, though, because you'd need to emulate the drive heads moving on the platters
> 
> As first step - "LED" indicator in the GUI (title bar or OSD) linked to HDD read/write actions. The noise part is more difficult - as you say emulating movement of drive heads is overkill - maybe playing a few recordings ("seeking", "heavy write", "light read", etc.) instead...

Yeah, we might play around with this once we have a basic OSD working. At least the LED support.

**Pull request merged**